acme/http.go

```diff
@@ -33,9 +34,11 @@ func httpHead(url string) (resp *http.Response, err error) {
 	req.Header.Set("User-Agent", userAgent())
 
 	resp, err = http.DefaultClient.Do(req)
-	if resp.Body != nil {
-		resp.Body.Close()
+	if err != nil {
+		return resp, err
 	}
+	io.Copy(ioutil.Discard, resp.Body)
```

Comment:
ha really just habit =). 

Technically these are HEAD requests so there shouldn't be any body, but in general, you must consume any body that's present before calling Close for the connection can be re-used by the underlying transport.